kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemViewKeyInputFormat.java

```diff
@@ -71,37 +69,38 @@
   public FileSystemViewKeyInputFormat(FileSystemDataset<E> dataset,
       Configuration conf) {
     this.dataset = dataset;
+    this.view = null;
     LOG.debug("Dataset: {}", dataset);
 
     Format format = dataset.getDescriptor().getFormat();
 
-    boolean isSpecific = SpecificRecord.class.isAssignableFrom(dataset.getType());
-
-    if (Formats.AVRO.equals(format)) {
-      setModel.invoke(conf,
-          DataModelUtil.getDataModelForType(dataset.getType()).getClass());
-
-      // Use the reader's schema type if provided.
-      if (isSpecific) {
-
-        conf.set(AVRO_SCHEMA_INPUT_KEY,
-            SpecificData.get().getSchema(dataset.getType()).toString());
-      }
-    } else if (Formats.PARQUET.equals(format)) {
-
-      // Use the reader's schema type if provided.
-      if (isSpecific) {
-
-        AvroReadSupport.setAvroReadSchema(conf,
-            SpecificData.get().getSchema(dataset.getType()));
+    if (!DataModelUtil.isGeneric(dataset.getType())) {
```

Comment:
I don't think this should only happen if the data model is not generic. If it is generic, we can still set the data model and schema correctly.